greenheart/tools/eco/hydrogen_mgmt.py

```diff
@@ -294,6 +298,7 @@ def run_h2_storage(
             hydrogen_storage_capacity_kg, hydrogen_storage_duration_hr, hydrogen_storage_soc = hydrogen_storage_capacity(electrolyzer_physics_results['H2_Results'], greenheart_config['electrolyzer']['rating'], hydrogen_storage_demand)
```

Comment:
I think it looks like we have two separate sets of logic around sizing, pressurized tower and demand/type. I think we should combine these sets of logic into a single if/else set starting with type for zero storage capacity, then checking demand, from max turbine storage, and finally using hour and max fill rate as default if the others are not specified. Eventually this should have the sizing type as a variable rather than potentially conflicting inputs, but for now we can at least simplify/clarify the logic

Reply:
I've changed the logic so that we have a single if/elif/else statement around sizing and then a signle if/elif/else statement around h2 storage technology type. I moved the logic around if on a turbine to inside the specific technology (pressure vessel and turbine) and included a check up front for illegal technology choices based on turbine location selection.